packages/go/ein/ad.go

```diff
@@ -40,13 +41,59 @@ func ConvertObjectToNode(item IngestBase, itemType graph.Kind) IngestibleNode {
 		itemProps = make(map[string]any)
 	}
 
+	if itemType == ad.Domain {
+		convertInvalidDomainProperties(itemProps)
+	}
+
 	return IngestibleNode{
 		ObjectID:    item.ObjectIdentifier,
 		PropertyMap: itemProps,
 		Label:       itemType,
 	}
 }
 
+func convertInvalidDomainProperties(itemProps map[string]any) {
+	convertStringPropertyToInt(itemProps, "machineaccountquota")
+	convertStringPropertyToInt(itemProps, "minpwdlength")
+	convertStringPropertyToInt(itemProps, "pwdproperties")
+	convertStringPropertyToInt(itemProps, "pwdhistorylength")
+	convertStringPropertyToInt(itemProps, "lockoutthreshold")
+
+	if rawProperty, ok := itemProps["expirepasswordsonsmartcardonlyaccounts"]; ok {
```

Comment:
Could this logic be wrapped in a convertStringPropertyToBool and used mirroring the convertStringPropertyToInt

Reply:
Or, as a step further abstract this into a shared method that you pass a conversion function.
convertProperty(itemProps, "machineaccountquota", stringToInt)
convertProperty(itemProps, "expirepasswordsonsmartcardonlyaccounts", stringToBool)

Reply:
I didn't abstract to another function because we only use it once, so it felt kind of pointless. The other function is re-used several times.